docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -283,269 +725,152 @@ quarkus.oidc.credentials.secret=secret
+==== Expiration via Session management
 
-[source, java]
-----
-import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.enterprise.event.Observes;
+By default, logout is based on the expiration time of the ID token issued by the OIDC provider. When the ID token expires, the current user session at the Quarkus endpoint is invalidated and the user is redirected to the OIDC provider again to authenticate. 
+If the session at the OIDC provider is still active, users are automatically re-authenticated without needing to provide their credentials again.
 
-import io.quarkus.oidc.IdTokenCredential;
-import io.quarkus.oidc.SecurityEvent;
-import io.quarkus.security.identity.AuthenticationRequestContext;
-import io.vertx.ext.web.RoutingContext;
+The current user session can be automatically extended by enabling the `quarkus.oidc.token.refresh-expired` property. 
+If set to `true`, when the current ID token expires, a refresh token grant will be used to refresh the ID token as well as access and refresh tokens.
 
-@ApplicationScoped
-public class SecurityEventListener {
+[TIP]
+--
+If you have a xref:security-oidc-bearer-token-authentication-concept.adoc#single-page-applications[single page application for service applications] where your OIDC provider script such as `keycloak.js` is managing an authorization code flow then that script will also control the SPA authentication session lifespan.
+--
 
-    public void event(@Observes SecurityEvent event) {
-        String tenantId = event.getSecurityIdentity().getAttribute("tenant-id");
-        RoutingContext vertxContext = event.getSecurityIdentity().getAttribute(RoutingContext.class.getName());
-        vertxContext.put("listener-message", String.format("event:%s,tenantId:%s", event.getEventType().name(), tenantId));
-    }
-}
-----
+If you work with a Quarkus OIDC `web-app` application, then it is the Quarkus OIDC code authentication mechanism that is managing the user session lifespan.
 
-=== Single-page applications
+To use the refresh token, you should carefully configure the various lifespan of the ID token, the refresh token, the cookie, etc so they work in tandem.
+The so called session age should be longer than the ID token lifespan and close to or equal to the refresh token lifespan.
 
-Check if implementing SPAs the way it is suggested in the xref:security-oidc-bearer-token-authentication-concept.adoc#single-page-applications[Single-page Applications for Service Applications] section can meet your requirements.
+The session age is calculated by adding the lifespan value of the current IDToken and the values of the `quarkus.oidc.authentication.session-age-extension` and `quarkus.oidc.token.lifespan-grace` properties. Of the last two properties only `quarkus.oidc.authentication.session-age-extension` should be used to significantly extend the session lifespan if required since `quarkus.oidc.token.lifespan-grace` is only meant for taking some small clock skews into consideration.
```

Comment:
```suggestion
You calculate the session age by adding the lifespan value of the current ID token and the values of the `quarkus.oidc.authentication.session-age-extension` and `quarkus.oidc.token.lifespan-grace` properties. 
[NOTE]
--
You use only the `quarkus.oidc.authentication.session-age-extension` property to significantly extend the session lifespan, if required. 
You use the `quarkus.oidc.token.lifespan-grace` property only for taking some small clock skews into consideration.
--

Reply:
@sheilamjones making it a `TIP` makes sense. The clock skew is a difference in clock setups on various computers, it is a well known technical term for it, so I suppose we can leave it as is